keymanager-api-common/src/api.rs

```diff
@@ -98,25 +99,34 @@ pub struct ReplicateResponse {
 
 /// Request runtime/key pair id tuple.
 #[derive(Clone, Default, cbor::Encode, cbor::Decode)]
-pub struct RequestIds {
+pub struct RuntimeKeyRequest {
     /// Runtime ID.
     pub runtime_id: Namespace,
     /// Key pair ID.
     pub key_pair_id: KeyPairId,
 }
 
-impl RequestIds {
+impl RuntimeKeyRequest {
     pub fn new(runtime_id: Namespace, key_pair_id: KeyPairId) -> Self {
         Self {
             runtime_id,
             key_pair_id,
         }
     }
+}
 
-    pub fn to_cache_key(&self) -> Vec<u8> {
-        let mut k = self.runtime_id.as_ref().to_vec();
-        k.extend_from_slice(self.key_pair_id.as_ref());
-        k
+/// Request runtime/key pair id tuple.
+#[derive(Clone, Default, cbor::Encode, cbor::Decode)]
+pub struct EphemeralKeyRequest {
```

Comment:
This also needs to be runtime-scoped similar to regular keys as a key manager can serve multiple runtimes and they should have distinct keys.